### PR TITLE
you may not shovestun batoned people

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

**_you_** may not shovestun people that were batoned 4 seconds ago or less
(others still can go gang up on people or something)

https://github.com/tgstation/tgstation/assets/70376633/709801a9-3c45-4859-a2f6-c7c28bc715d6

## Why It's Good For The Game

batons have a delay to not instantly stamcrit someone and baton+shoving someone to achieve vaguely the same thing seems like going around that

also that is just not fun to play against and its pretty cheap

with this change you will have gang up on people (socializing??? in my RP game???) to stunlock or just wait for the cooldown to expire to stamcrit the victim which is miles better than an almost instant paralyze

## Changelog
:cl:
balance: the person that batons another person may not shovestun that person for 4 seconds
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
This has exactly the same problem as #82231 in that "this identical knockdown works differently from other ones" is an unintuitive and completely invisible mechanic.

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -705,7 +705,7 @@
 			log_combat(src, target, "shoved", "knocking them down[weapon ? " with [weapon]" : ""]")
 			return
 
-	if(shove_flags & SHOVE_CAN_KICK_SIDE) //KICK HIM IN THE NUTS
+	if((shove_flags & SHOVE_CAN_KICK_SIDE) && !HAS_TRAIT_FROM(target, TRAIT_IWASBATONED, REF(src))) //KICK HIM IN THE NUTS
```
Comment (MrMelbert) [inline code/modules/mob/living/living_defense.dm:708]: This check should be in `/mob/living/carbon/get_shove_flags`, IE, it should not return `SHOVE_CAN_KICK_SIDE` if they have the trait

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by mc-oofert** [comment]
I can't mark as draft from my phone

**Comment by Metekillot** [comment]
I don't think shovestun should be a thing.

But, since it's here right now:


This seems like an arbitrary distinction. Why do you think that baton knockdowns should exclusively not allow shove stuns?

**Comment by Livrah** [comment]
Shove stun removal testmerge was the best time to play.

**Comment by Sakamoto4ka** [comment]
Finally

**Comment by mc-oofert** [comment]
> Shove stun removal testmerge was the best time to play. 

Worst

**Comment by Deultus** [comment]
It may seem strange, but I would like it to be in game. Show me someone who likes to lie on the floor from a push, plus this is a certificate for weakening heretics

**Comment by SimplySmiley123** [comment]
> It may seem strange, but I would like it to be in game. Show me someone who likes to lie on the floor from a push, plus this is a certificate for weakening heretics

The main problem is how it's gonna be told to the use of the baton that they can't do it, that it's not a possibility at all.
Is it gonna be in the baton's description?
A special tooltip?
Or just silent with nothing to tell this exists?

**Comment by mc-oofert** [comment]
> It may seem strange, but I would like it to be in game. Show me someone who likes to lie on the floor from a push, plus this is a certificate for weakening heretics

mansus grasp does not apply the batoned trait

**Comment by mc-oofert** [comment]
https://github.com/tgstation/tgstation/assets/70376633/709801a9-3c45-4859-a2f6-c7c28bc715d6

**Comment by Higgin** [comment]
not a good idea and not even the most potent combo with batons. especially troubling with innate attacks and people who know how to rest to foil the disarm.

File: code/modules/mob/living/living_defense.dm [hunk 2]
```diff
@@ -705,7 +705,7 @@
 			log_combat(src, target, "shoved", "knocking them down[weapon ? " with [weapon]" : ""]")
 			return
 
-	if(shove_flags & SHOVE_CAN_KICK_SIDE) //KICK HIM IN THE NUTS
+	if((shove_flags & SHOVE_CAN_KICK_SIDE)) //KICK HIM IN THE NUTS
```
Comment (MrMelbert) [inline code/modules/mob/living/living_defense.dm:708]:
```suggestion
	if(shove_flags & SHOVE_CAN_KICK_SIDE) //KICK HIM IN THE NUTS
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -699,6 +699,9 @@
 	. = ..()
 	. |= SHOVE_CAN_STAGGER
 	if(IsKnockdown() && !IsParalyzed())
+		if(HAS_TRAIT_FROM(src, TRAIT_IWASBATONED, REF(shover))) // so we can do a balloon alert
+			balloon_alert(shover, "tired from baton use to kick onto side!")
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/carbon_defense.dm:703]:
```suggestion
			balloon_alert(shover, "can't kick after batonning!")
```

**Comment by GuillaumePrata** [comment]
At this point wouldn't it just be easier to remove the knockdown of the baton on the first hit?

If they disarm and apply a slow effect it is basically the same thing outside of some fringe interactions, and it would be a lot easier to understand why you can't wall shove combo people.

**Comment by Jacquerel** [comment]
> At this point wouldn't it just be easier to remove the knockdown of the baton on the first hit?
> 
> If they disarm and apply a slow effect it is basically the same thing outside of some fringe interactions, and it would be a lot easier to understand why you can't wall shove combo people.

yeah I can see why people are going for these more bespoke solutions (it's easier and they don't want to worry about the balance of other features) but I think anything which adds conditional effects like this is inadvisable
either rework batons or add a new status effect with its own visuals (or more likely, add visuals to the existing one which make it obvious that you are more vulnerable) so that people can tell which they are being effected by

**Review by Ghommie — APPROVED** [review]

**Comment by optimumtact** [comment]
this is surely bait that peopel keep making this PR.

**Comment by Sakamoto4ka** [comment]
first pr when roblocks doesn't bring his baton nerf picture

**Pull request closed without merging**